### ISSUE TITLE
[PEFT] Bug combining some loras from civitai using peft

### ISSUE THREAD

**Issue by jaimereyley** [huggingface/diffusers#5395]
### Describe the bug

Inference bug when combining some LORAs from Civitai using PEFT. 

Google colab with reproduction: https://colab.research.google.com/drive/1bpVehy9NQDEIRt5JeYfKGFlw7KZdlDrm?usp=sharing

in peft/tuners/lora/layer.py
(self.active_adapters contain adapters not present in self.lora_alpha)

### Reproduction

!pip install -q transformers accelerate
!pip install -q git+https://github.com/huggingface/peft.git
!pip install -q git+https://github.com/huggingface/diffusers.git

!wget -O oil_painting.safetensors https://civitai.com/api/download/models/163652

from diffusers import DiffusionPipeline
import torch

pipe_id = "stabilityai/stable-diffusion-xl-base-1.0"
pipe = DiffusionPipeline.from_pretrained(pipe_id, torch_dtype=torch.float16).to("cuda")

pipe.load_lora_weights("CiroN2022/toy-face", weight_name="toy_face_sdxl.safetensors", adapter_name="toy")
pipe.load_lora_weights("./", weight_name="oil_painting.safetensors", adapter_name="oil_painting")
pipe.set_adapters(["oil_painting", "toy"], adapter_weights=[0.5, 1.0])
print(pipe.get_active_adapters())

prompt = "toy_face of a hacker with a hoodie, painting"
image = pipe(prompt, num_inference_steps=20, cross_attention_kwargs={"scale": 1.0}, generator=torch.manual_seed(0)
).images[0]
image

### Logs

```shell
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-8-45f9c7879915> in <cell line: 2>()
      1 prompt = "toy_face of a hacker with a hoodie, painting"
----> 2 image = pipe(
      3     prompt,
      4     num_inference_steps=20,
      5     cross_attention_kwargs={"scale": 1.0},

5 frames
/usr/local/lib/python3.10/dist-packages/peft/tuners/lora/layer.py in unscale_layer(self)
    163     def unscale_layer(self) -> None:
    164         for active_adapter in self.active_adapters:
--> 165             alpha = self.lora_alpha[active_adapter]
    166             r = self.r[active_adapter]
    167             self.scaling[active_adapter] = alpha / r

KeyError: 'toy'
```

### System Info

- `diffusers` version: 0.22.0.dev0
- Platform: Linux-5.10.0-23-cloud-amd64-x86_64-with-glibc2.31
- Python version: 3.11.4
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.17.3
- Transformers version: 4.34.0
- Accelerate version: 0.23.0
- xFormers version: not installed
- Using GPU in script?: Nvidia T4
- Using distributed or parallel set-up in script?: NO

### Who can help?

@sayakpaul
@younesbelkada
@pacman100

**Comment by younesbelkada** [comment]
Thanks very much for the repro, this will be super helpful for identifying the bug ! cc @pacman100 as well

**Comment by pacman100** [comment]
Hello @jaimereyley, the above PR should fix the issue. However, with the stack trace, I think that the adapter weights are being reset when it is passed by  `cross_attention_kwargs`. cc @younesbelkada and @sayakpaul to fix that and docs related to it in another PR. 

Now, I do the following:
```
!wget -O oil_painting.safetensors https://civitai.com/api/download/models/163652
from diffusers import DiffusionPipeline
import torch

pipe_id = "stabilityai/stable-diffusion-xl-base-1.0"
pipe = DiffusionPipeline.from_pretrained(pipe_id, torch_dtype=torch.float16).to("cuda")
pipe.load_lora_weights("CiroN2022/toy-face", weight_name="toy_face_sdxl.safetensors", adapter_name="toy")
pipe.load_lora_weights("./", weight_name="oil_painting.safetensors", adapter_name="oil_painting")
pipe.set_adapters(["oil_painting", "toy"], adapter_weights=[0.5, 1.0])
prompt = "toy_face of a hacker with a red hoodie in the style of oil painting. starry night in the background"
image = pipe(
    prompt,
    num_inference_steps=20,
#     cross_attention_kwargs={"scale": 1.0}
).images[0]
image
```

The output:
![download (81)](https://github.com/huggingface/diffusers/assets/13534540/f0702814-c7cb-4ac4-a689-c061c0b5feeb)

**Comment by younesbelkada** [comment]
This is actually fixed by #5414 - closing!
